# Target repository: frameworks/base
#
# Framework side of the offload path.
#
# Paths below are relative to the target repository, NOT to the Android
# source root. Apply with:  git -C <target repository> apply <this file>
#
# Publication edition. Do not mix with another edition of these patches.
diff --git a/media/java/android/media/AudioSystem.java b/media/java/android/media/AudioSystem.java
index 442fc38d13d9..aff7c7ea7103 100644
--- a/media/java/android/media/AudioSystem.java
+++ b/media/java/android/media/AudioSystem.java
@@ -250,6 +250,8 @@ public class AudioSystem
     /** @hide */
     public static final int AUDIO_FORMAT_LC3            = 0x2B000000;
     /** @hide */
+    public static final int AUDIO_FORMAT_APTX_ADAPTIVE  = 0x27000000;
+    /** @hide */
     public static final int AUDIO_FORMAT_OPUS           = 0x08000000;
     /** @hide */
     public static final int AUDIO_FORMAT_OPUS_HI_RES    = 0x08000001;
@@ -262,6 +264,7 @@ public class AudioSystem
             AUDIO_FORMAT_SBC,
             AUDIO_FORMAT_APTX,
             AUDIO_FORMAT_APTX_HD,
+            AUDIO_FORMAT_APTX_ADAPTIVE,
             AUDIO_FORMAT_LDAC,
             AUDIO_FORMAT_LC3,
             AUDIO_FORMAT_OPUS,
@@ -292,6 +295,8 @@ public class AudioSystem
 
     //TODO b/396350294 : remove when BluetoothLeCodecConfig.SOURCE_CODEC_TYPE_OPUS_HI_RES is public
     private static final int BLUETOOTH_LE_AUDIO_CODEC_CONFIG_SOURCE_CODEC_TYPE_OPUS_HI_RES = 2;
+    // aptX Adaptive is an extended native A2DP codec and is not part of the public codec API.
+    private static final int BLUETOOTH_CODEC_CONFIG_SOURCE_CODEC_TYPE_APTX_ADAPTIVE = 7;
     /**
      * @hide
      * Convert audio format enum values to Bluetooth codec values
@@ -303,6 +308,8 @@ public class AudioSystem
             case AUDIO_FORMAT_SBC: return BluetoothCodecConfig.SOURCE_CODEC_TYPE_SBC;
             case AUDIO_FORMAT_APTX: return BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX;
             case AUDIO_FORMAT_APTX_HD: return BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX_HD;
+            case AUDIO_FORMAT_APTX_ADAPTIVE:
+                return BLUETOOTH_CODEC_CONFIG_SOURCE_CODEC_TYPE_APTX_ADAPTIVE;
             case AUDIO_FORMAT_LDAC: return BluetoothCodecConfig.SOURCE_CODEC_TYPE_LDAC;
             case AUDIO_FORMAT_LC3: return BluetoothCodecConfig.SOURCE_CODEC_TYPE_LC3;
             case AUDIO_FORMAT_OPUS: return BluetoothCodecConfig.SOURCE_CODEC_TYPE_OPUS;
